hack/lib/golang.sh

```diff
@@ -521,7 +521,10 @@ kube::golang::setup_env() {
   kube::golang::create_gopath_tree
   export GOPATH="${KUBE_GOPATH}"
 
-  export GOCACHE="${KUBE_GOPATH}/cache"
+  # If these are not set, set them now.  This ensures that any subsequent
+  # scripts we run (which may call this function again) use the same values.
+  export GOCACHE="${GOCACHE:-"${KUBE_GOPATH}/cache/build"}"
+  export GOMODCACHE="${GOMODCACHE:-"${KUBE_GOPATH}/cache/mod"}"
```

Comment:
GOMODCACHE already defaults to being under GOPATH... curious why we need to relocate it

Reply:
this was already answered in another review... this makes sure things we run in worktrees get to reuse already cached module content in the original clone location

Reply:
This is more about the "if these are not set" - same comment as other PR (this hunk was needed in both, but will rebase out, whichever lands first).